### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by Pastelcio** [vaibhavvikas/factorio-mod-downloader#5] [Ideas]
Hello, I've got two suggestions for the Factorio mod downloader.
The first suggestion is to disable the "Directory is not empty, do you want to continue and overwrite?" message. It can be annoying to download a lot of mods and see this every single time.
The second suggestion is to update existing mods by checking their version (info.json) and comparing it to the most recent one on the site. <-This one can be difficult, and I'm not even sure if it's possible. I'm also not sure how you'll make it find the link to update because some mods don't include it in the info.json, my guess is to save the link when you download the mod, this would solve the problem. but as the title says, it's just a suggestion; you don't have to add it <3

**Comment by vaibhavvikas** [comment]
Hi @Pastelcio Thanks for the suggestions.

1. The first option, i.e. to disable the overwrite message, I agree with you that it can be annoying, I believe the better solution will be to display the warning message only once i.e. during the first download only rather than removing it entirely. Reason being many users might mistakenly select the wrong folder, and it might ruin the existing mods or other files.

2. Getting version number is not a big deal, as per factorio mod requirements, the zip name should be matching the info.json inside it. So basically you can get version number by the mod zip name itself. Challenge is getting the mod url based on its name. I will see how it can be done. Since in the actual Factorio you get the version info and everything. It is a big change. Also, another thing lets say you selected a folder which already have other mods, which you downloaded separately, we need to support that as well, so saving the link will not work on these scenarios.

I will definitely add the first feature in the next update, regarding second change it will take a while. Will work on it when I will get time.

**Comment by Youssef-Hamoda** [comment]
Hey, I've got a suggestion as the devs start to crack down on things like this.

I nearly got locked out my game when I tried to delete a mod I didn't want anymore. I get locked when I try to sync my loads with the mods I have and if there's any incongruency it just says mod portal only available if you own the game. If we could get a way to delete and a way to sync loads through this tool.

Would be great.

**Reply by vaibhavvikas** [reply]
Hi @Youssef-Hamoda Thanks for the suggestion. The issue is, the mod downloader is entirely a separate downloader, which downloads the latest mod version and its required dependencies. It doesn't interact with the main factorio entirely and doesn't knows which factorio version is installed etc. Your suggestion is very good however its implementation will require to interact with factorio application and will need information such as which version is installed, and then checking each mod one by one and trying to figure out if it will work or not. Which in short is very complicated to implement. I will definitely consider it.

**Comment by Derjyn** [comment]
I second the suggestion for disabling the pop-up related to mod folder. Easy solution, make a config file. Optional message display for empty directory, and also download completion.

On that note, where are options currently saved? Right now that would probably just be the download directory, but curious if this is being saved to registry, appdata, etc.

I keep a tight ship, and I loathe applications that reach outside of specified directories. To be more specific: I feel applications should universally only keep *.ini files, logs, those sort of files - in the install/working directory. And also garbage collection on uninstall. Software nowadays is so damned sloppy, leaving garbage all over! Obviously this isn't specific to this handy tool, just a general observation so that if you do expand features (not that FMD needs many more features at all), please take care to keep it neat and tidy ;)

**Comment by RebornSplash60official** [comment]
One thing( well two things actually) I would like to see in an update is the ability to choose witch version of the mod to download along with the game version.

**Reply by Erumaaro** [reply]
yeah, set game version -> downloads latest compatible version of target mod and correct version of dependencies, recursively.

**Reply by RebornSplash60official** [reply]
Ok

**Reply by vaibhavvikas** [reply]
Thanks giys for the suggestions. I have implemented all these features. You can select the factorio version youre using and it will automatically upgrade or downgrade the model specific versions. It will also give you the option to remove incompatible mods.

Further there is also a search functionality to directly get the mod directly from the app rather than searching and manually pasting ids.
